spec/services/workflow_delete_service_spec.rb

```diff
@@ -0,0 +1,21 @@
+RSpec.describe WorkflowDeleteService do
+  let(:workflow) { create(:workflow, :group_refs => [990, 991]) }
+  let(:aps) { instance_double(AccessProcessService) }
+
+  subject { described_class.new(workflow.id) }
+
+  context 'when update' do
+    before do
+      allow(AccessProcessService).to receive(:new).and_return(aps)
+      allow(aps).to receive(:remove_resource_from_groups)
+    end
+
+    it 'returns status code 204' do
+      Insights::API::Common::Request.with_request(RequestSpecHelper.default_request_hash) do
+        subject.cleanup
+
+        expect(Workflow.find_by(:id => workflow.id)).to be_nil
```

Comment:
also need to assert the acls being removed, and tag links removed.

Reply:
`acls` is stub mockup. Tests for tag links will add later.